### PR TITLE
prefix for broadinstitute/cromwell#375

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 1]
```diff
@@ -114,6 +116,23 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WdlValue] {
...
   }
 
+  def prefix(params: Seq[Try[WdlValue]]): Try[WdlArray] = {
+    def extractTwoArguments: Try[(WdlValue, WdlValue)] = params.size match {
+      case 2 => (params.head |@| params.tail.head) map { (_, _) }
+      case n => Failure(new UnsupportedOperationException(s"prefix() expects two parameters but got $n"))
+    }
+
+    val makePrefixedString = (prefixString: WdlValue, strings: WdlValue) => (prefixString, strings) match {
+      case (WdlString(p), WdlArray(WdlArrayType(WdlStringType), ss)) =>
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: Might be nice to support prefixing any primitive array type with:
```
case (WdlString(p), WdlArray(WdlArrayType(x), ss)) if x.isInstanceOf[WdlPrimitiveType] =>
```

NB: we'd still want Strings as output, obvs...

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 2]
```diff
@@ -114,6 +116,23 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WdlValue] {
...
+        val result = ss map { s => WdlString(p + s.valueString) }
+        Success(WdlArray(WdlArrayType(WdlStringType), result))
+      case (x, y) =>
+        Failure(new UnsupportedOperationException(s"prefix() expects a String and Array[String] parameters but got (${prefixString.wdlType.toWdlString}, ${strings.wdlType.toWdlString})"))
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: Inconsistent format, would prefer `The function prefix expects arguments (X, Y) but got (X', Y')`

**Comment by cjllanwarne** [comment]
ToL comments, otherwise 👍

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: mokay but you just volunteered to review the updated docs 😛

**Comment by geoffjentry** [comment]
👍

**Pull request merged**